lib/streamlit/elements/vega_charts.py

```diff
@@ -790,780 +893,134 @@ def altair_chart(
-        pass
-
-
-def _convert_col_names_to_str_in_place(
-    df: pd.DataFrame,
-    x_column: str | None,
-    y_column_list: list[str],
-    color_column: str | None,
-    size_column: str | None,
-) -> tuple[str | None, list[str], str | None, str | None]:
-    """Converts column names to strings, since Vega-Lite does not accept ints, etc."""
-    import pandas as pd
-
-    column_names = list(df.columns)  # list() converts RangeIndex, etc, to regular list.
-    str_column_names = [str(c) for c in column_names]
-    df.columns = pd.Index(str_column_names)
-
-    return (
-        None if x_column is None else str(x_column),
-        [str(c) for c in y_column_list],
-        None if color_column is None else str(color_column),
-        None if size_column is None else str(size_column),
-    )
-
-
-def _parse_generic_column(
-    df: pd.DataFrame, column_or_value: Any
-) -> tuple[str | None, Any]:
-    if isinstance(column_or_value, str) and column_or_value in df.columns:
-        column_name = column_or_value
-        value = None
-    else:
-        column_name = None
-        value = column_or_value
-
-    return column_name, value
-
-
-def _parse_x_column(df: pd.DataFrame, x_from_user: str | None) -> str | None:
-    if x_from_user is None:
-        return None
-
-    elif isinstance(x_from_user, str):
-        if x_from_user not in df.columns:
-            raise StreamlitColumnNotFoundError(df, x_from_user)
-
-        return x_from_user
-
-    else:
-        raise StreamlitAPIException(
-            "x parameter should be a column name (str) or None to use the "
-            f" dataframe's index. Value given: {x_from_user} "
-            f"(type {type(x_from_user)})"
-        )
-
-
-def _parse_y_columns(
-    df: pd.DataFrame,
-    y_from_user: str | Sequence[str] | None,
-    x_column: str | None,
-) -> list[str]:
-    y_column_list: list[str] = []
-
-    if y_from_user is None:
-        y_column_list = list(df.columns)
-
-    elif isinstance(y_from_user, str):
-        y_column_list = [y_from_user]
-
-    elif type_util.is_sequence(y_from_user):
-        y_column_list = list(str(col) for col in y_from_user)
-
-    else:
-        raise StreamlitAPIException(
-            "y parameter should be a column name (str) or list thereof. "
-            f"Value given: {y_from_user} (type {type(y_from_user)})"
-        )
+    @gather_metrics("vega_lite_chart")
+    def vega_lite_chart(
+        self,
+        data: Data = None,
+        spec: dict[str, Any] | None = None,
+        use_container_width: bool = False,
+        theme: Literal["streamlit"] | None = "streamlit",
+        **kwargs: Any,
+    ) -> DeltaGenerator:
+        """Display a chart using the Vega-Lite library.
 
-    for col in y_column_list:
-        if col not in df.columns:
-            raise StreamlitColumnNotFoundError(df, col)
+        Parameters
+        ----------
+        data : pandas.DataFrame, pandas.Styler, pyarrow.Table, numpy.ndarray, Iterable, dict, or None
```

Comment:
nit: the TypeUnion says it can also be a `Series` and an `Index`. Should this be included in the comment?

Reply:
Yep, but also a bit out of scope since there are a lot of types missing and it will require a bit of rethinking and unification on the docs side: See this comment: https://github.com/streamlit/streamlit/pull/8588#issuecomment-2084950230